static/src/js/components/AnnouncementList.react.js

```diff
@@ -0,0 +1,158 @@
+/* @file announcements.js
+ * @brief Creates the React components associated with announcements, and
+ * exports the ones we want to create on the fly.
+ *
+ * @author Oscar Bezi (bezi@scottylabs)
+ */
+'use strict';
+
+var React = require('react');
+var moment = require('moment');
+
+var Input = require('react-bootstrap').Input;
+var ListGroup = require('react-bootstrap').ListGroup;
+var Well = require('react-bootstrap').Well;
+var Button = require('react-bootstrap').Button;
+var Glyphicon = require('react-bootstrap').Glyphicon;
+
+var Announcement = require('./Announcement.react');
+
+var Store = require('../stores/AnnouncementStore');
+var Actions = require('../actions/AnnouncementActions');
+
+var api = require('../api/announcements');
+
+class AnnouncementList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      'announcements': [],
+      'value': '',
+    };
+
+    // Bind handlers.
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.handleKeypress = this.handleKeypress.bind(this);
+    this.updateState = this.updateState.bind(this);
+  }
+
+  /* @brief Enables the timer that updates the element. */
+  componentWillMount() {
+    Store.register(this.updateState);
+    this.updateState();
+  }
+
+  /* @brief Cleans up the timer that updates the element. */
+  componentWillUnmount() {
+    Store.deregister(this.updateState);
+  }
+
+  /* @brief Gets state from the store. */
+  updateState() {
+    this.setState({
+      'announcements': Store.get(),
+      'value': this.state.value,
+    });
+  }
+
+  /* @brief Handles form submission. */
+  handleFormSubmit(e) {
+    e.preventDefault();
+    if (this.state.value !== '') {
+      Actions.create(this.state.value);
+      this.setState({
+        'announcements': this.state.announcements,
+        'value': '',
+      });
+    }
+  }
+
+  /* @brief Handles keypresses. */
+  handleKeypress() {
+    if (this.refs.input.getValue().length <= api.maxLength) {
+      this.setState({
+        'announcements': this.state.announcements,
+        'value': this.refs.input.getValue(),
+      });
+    }
+  }
+
+  /* @brief Returns whether the the input has a valid input. */
+  validationState() {
+    if (this.state.value === '') {
+      return;
+    }
+
+    return this.state.value.length <= api.maxLength ? 'success' : 'error';
+  }
+
+  /* @brief Returns the built form. */
+  buildForm() {
+    var button = <Button type="submit"><Glyphicon glyph="upload" /></Button>;
+    var form = (
+      <form onSubmit={this.handleFormSubmit}>
+      <Input
```

Comment:
There are a number of cases where nested JSX isn't properly indented. Rather than leave a comment on each one, just make sure it gets addressed eventually.